### PR TITLE
Added documentation for smart-content categories

### PR DESCRIPTION
**Pull request by wachterjohannes**
Main PR: sulu-io/sulu#1606

### CODE REVIEW

File: bundles/admin/javascript-hooks/defaults.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
+Defaults
+========
+
+The default-hook merges and prepares the options, translations and templates
+automatically. This normalizes and centralize the need functionality of this
+needs.
```
Comment (danrot) [inline bundles/admin/javascript-hooks/defaults.rst]: The last sentence sounds a bit strange, should probably be rewritten.

File: bundles/admin/javascript-hooks/events.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
+Events
+======
+
+The event-hook manages events and creates callable functions to throw or catch
+aura events. It capsules the createEventName function to build event-names with
```
Comment (danrot) [inline bundles/admin/javascript-hooks/events.rst]: I would use `encapsulate`, not sure if `capsule` can be used like this.

File: cookbook/smart-content-data-provider.rst [hunk 3]
```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
...
-objects and for the website with array and entity access. Additionally the
-DataProvider returns a configuration object to enable or disable features.
+functions to resolve the configured filters for the backend API with
+standardized objects and for the website with array and entity access.
+Additionally the DataProvider returns a configuration object to enable or
+disable features.
 
-For example how to implement it for ORM based entities see the Pull-Request for
-the `Account DataProvider`_.
+For ORM DataProvider there exists some abstraction layer. This layer provides the
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: `There exists an abstraction layer for ORM DataProviders.

File: cookbook/smart-content-data-provider.rst [hunk 4]
```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
...
+implementation of basic DataProvider functionality and Database query.
 
-.. _Account DataProvider: https://github.com/sulu-io/sulu/pull/1517
+If you want to create a DataProvider for the ExampleEntity you have todo
+following steps.
+
+1. Repository
+~~~~~~~~~~~~~
+
+Implement Interface `DataProviderRepositoryInterface` to provide needed function
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: The repository has to implement the `DataProviderRepositoryInterface` and provide the `findByFilters` function.

File: cookbook/smart-content-data-provider.rst [hunk 5]
```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
...
+findByFilters. If the default implementation is enough for you you can include
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: If the default implementation is good enough, you can include...

File: cookbook/smart-content-data-provider.rst [hunk 6]
```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
...
+the trait `DataProviderRepositoryTrait` which needs the functions
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: `DataProviderRepositoryTrait`, which, ...

File: cookbook/smart-content-data-provider.rst [hunk 7]
```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
...
+`createQueryBuilder` (is default in repositories) and `appendJoins` there you
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: `appendJoins`, where you...

File: cookbook/smart-content-data-provider.rst [hunk 8]
```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
...
+        {
+            return $this->entity;
+        }
+    }
+
+3. DataProvider
+~~~~~~~~~~~~~~~
+
+Also the DataProvider is mostly abstracted by the SmartContent Component. The
+optimize the configuration you can disable or enable the form-elements to avoid
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: In the configuration you can disable or enable ...

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**